app/services/articles/analytics_updater.rb

```diff
@@ -0,0 +1,54 @@
+module Articles
+  class AnalyticsUpdater
+    def initialize(context = "default")
+      @context = context
+    end
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def call(user_id)
+      articles_to_check = Article.where(user_id: user_id, published: true)
```

Comment:
if you pass the user object to the `call` you can just do `articles_to_check = user.articles.published` here

Reply:
I have changed in that way and refactored a bit too. I had then to update the spec accordingly.